src/main/scala/com/github/potix2/spark/google/spreadsheets/DefaultSource.scala

```diff
@@ -13,21 +13,19 @@
  */
 package com.github.potix2.spark.google.spreadsheets
 
-import java.io.File
-
+import com.github.potix2.spark.google.spreadsheets.util.Credentials
+import com.google.auth.http.HttpCredentialsAdapter
 import org.apache.spark.sql.sources.{BaseRelation, CreatableRelationProvider, RelationProvider, SchemaRelationProvider}
 import org.apache.spark.sql.types.StructType
 import org.apache.spark.sql.{DataFrame, SQLContext, SaveMode}
 
 class DefaultSource extends RelationProvider with SchemaRelationProvider with CreatableRelationProvider {
-  final val DEFAULT_CREDENTIAL_PATH = "/etc/gdata/credential.p12"
 
-  override def createRelation(sqlContext: SQLContext, parameters: Map[String, String]) = {
+  override def createRelation(sqlContext: SQLContext, parameters: Map[String, String]): BaseRelation = {
     createRelation(sqlContext, parameters, null)
   }
 
-  private[spreadsheets] def pathToSheetNames(parameters: Map[String, String]): (String, String) = {
-    val path = parameters.getOrElse("path", sys.error("'path' must be specified for spreadsheets."))
+  private[spreadsheets] def pathToSheetNames(path: String): (String, String) = {
     val elems = path.split('/')
     if (elems.length < 2)
       throw new Exception("'path' must be formed like '<spreadsheet>/<worksheet>'")
```

Comment:
Yuk! Throwing exceptions!!!

Reply:
That was present in the original repository. This should be fixed in a separate PR.